lib/utils/couchclient/src/index.js

```diff
@@ -699,65 +738,84 @@ const couchclient = (part, uri, cons) => {
-const drop = (server, regex, cb) => {
-  if(/:/.test(server)) {
-    // Only do this on localhost or 127.0.0.1 for now as that's only for
-    // running our tests
-    const u = url.parse(server);
-    if(u.hostname !== 'localhost' && u.hostname != '127.0.0.1') {
-      debug('Server not on localhost, not deleting dbs on %s', server);
-      cb();
-      return;
-    }
+const isLocalServer = (server) => {
+  const u = url.parse(server);
+  return u.hostname === 'localhost' || u.hostname === '127.0.0.1';
+};
 
-    // List all remote dbs
-    debug('Getting list of all dbs matching %s on %s', regex, server);
-    request.get(':server/_all_dbs', {
-      server: server
-    }, (err, val) => {
-      if(err) {
-        cb(err);
-        return;
-      }
-      if(!val.body)
-        cb();
-      else
-        dropDBs(val.body, regex, server, {}, cb);
+// Drop all server databases that match the specified regex
+const dropServerDatabases = (server, regex, options, localOnly, cb) => {
+  debug('Deleting databases matching %s on %s', regex, server);
+
+  if (!localOnly || isLocalServer(server))
+    // List all databases
+    listDatabases(server, (err, dbs) => {
+      if (err)
+        return cb(err);
+
+      // Drop those databases from the list that match the regex
+      return dropListedDatabases(server, dbs, regex, options, cb);
     });
-  }
   else {
-    debug('In-memory DB %s detected; deleting dbs', server);
+    debug('Skipping non-local server %s', server);
+    cb(null, []);
+  }
+};
 
-    const globalStoreLocations = Object.keys(memdown('_$nonexistent$_')._store);
-    const prefix = '$' + (server ? server + '/' : '');
-    debug('Removing prefix %s from db names', prefix);
-    const dbNames = map(globalStoreLocations, (key) => key.replace(prefix, ''));
+// Drop all in-memory databases that match the specified regex
+const dropInMemoryDatabases = (server, regex, options, cb) => {
+  debug('Deleting in-memory databases matching %s on %s', regex, server);
 
-    debug('Found in-memory DBs %o', dbNames);
-    dropDBs(dbNames, regex, server, inMemoryOpt, cb);
-  }
+  // List all in-memory databases
+  const dbs = listInMemoryDatabases(server);
+
+  // Drop those databases from the list that match the regex
+  dropListedDatabases(server, dbs, regex, options, cb);
+};
+
+// Drop all databases that match the specified regex
+const dropDatabases = (server, regex, localOnly, cb) => {
+  const lcb = (err, deletedDatabases) => {
+    if (err)
+      return cb(err);
+
+    if(deletedDatabases.length > 0)
+      debug('Deleted databases: %o', deletedDatabases);
+    else
+      debug('No databases were deleted');
+    return cb(null, deletedDatabases);
+  };
+
+  if(/:/.test(server))
+    dropServerDatabases(server, regex, {}, localOnly, lcb);
+  else
+    dropInMemoryDatabases(server, regex, inMemoryOpt, lcb);
+};
+
+// Delete all partitions that match the specified regex
+const deletePartitions = (server, regex, cb) => {
```

Comment:
"Partition" is a unit that's used for partitioning. With Couch, partitions are always databases. With Mongo, however, partitions can be databases or collections, and `deletePartitions` will correctly determine what to delete in both cases. Therefore, the naming "partition" is important and since both clients should have the same API, both should have such a method.

The method `deletePartitions` is intentionally different from `drop` because:
* It always works with partitions, which might be different from databases (with Mongo)
* It deletes partitions also on remote servers, while `drop` deletes databases on local servers only.